src/fsharp/DependencyManager.Integration.fs

```diff
@@ -0,0 +1,182 @@
+| UnknownType
+
+type ReflectionDependencyManagerProvider(theType: Type, nameProperty: PropertyInfo, keyProperty: PropertyInfo, resolveDeps: MethodInfo, outputDir: string option) =
+    let instance = Activator.CreateInstance(theType, [|outputDir :> obj|])
+    let nameProperty     = nameProperty.GetValue >> string
+    let keyProperty      = keyProperty.GetValue >> string
+
+    static member InstanceMaker (theType: System.Type, outputDir: string option) =
+        match ReflectionHelper.getAttributeNamed theType dependencyManagerAttributeName,
+                ReflectionHelper.getInstanceProperty<string> theType namePropertyName,
+                ReflectionHelper.getInstanceProperty<string> theType keyPropertyName,
+                ReflectionHelper.getInstanceMethod<bool * string list * string list> theType [| typeof<string>; typeof<string>; typeof<string>; typeof<string seq>; typeof<string> |] resolveDependenciesMethodName
+                with
+        | None, _, _, _ -> None
+        | _, None, _, _ -> None
+        | _, _, None, _ -> None
+        | _, _, _, None -> None
+        | Some _, Some nameProperty, Some keyProperty, Some resolveDependenciesMethod ->
+            Some (fun () -> new ReflectionDependencyManagerProvider(theType, nameProperty, keyProperty, resolveDependenciesMethod, outputDir) :> IDependencyManagerProvider)
+
+    interface IDependencyManagerProvider with
+        member __.Name     = instance |> nameProperty
+        member __.Key      = instance |> keyProperty
+        member __.ResolveDependencies(scriptDir, mainScriptName, scriptName, packageManagerTextLines, tfm) =
+            let arguments = [| box scriptDir; box mainScriptName; box scriptName; box packageManagerTextLines; box tfm |]
+            resolveDeps.Invoke(instance, arguments) :?> _
+
+// Resolution Path = Location of FSharp.Compiler.Private.dll
+let assemblySearchPaths = lazy (
+    [
+        let assemblyLocation = typeof<IDependencyManagerProvider>.GetTypeInfo().Assembly.Location
+        yield Path.GetDirectoryName assemblyLocation
+        yield AppDomain.CurrentDomain.BaseDirectory
+    ])
+
+let enumerateDependencyManagerAssemblies compilerTools m =
+    getCompilerToolsDesignTimeAssemblyPaths compilerTools
+    |> Seq.append (assemblySearchPaths.Force())
+    |> Seq.collect (fun path ->
+        try
+            if Directory.Exists(path) then Directory.EnumerateFiles(path, dependencyManagerPattern)
+            else Seq.empty
+        with _ -> Seq.empty)
+    |> Seq.choose (fun path -> 
+        try
+            Some(AbstractIL.Internal.Library.Shim.FileSystem.AssemblyLoadFrom path)
+        with 
+        | e ->
+            let e = ReflectionHelper.stripTieWrapper e
+            warning(Error(FSComp.SR.couldNotLoadDependencyManagerExtension(path,e.Message),m))
+            None)
+    |> Seq.filter (fun a -> ReflectionHelper.assemblyHasAttribute a dependencyManagerAttributeName)
+
+let registeredDependencyManagers = ref None
```

Comment:
This reference is never written to, the assemblies and types seems to currently be scanned on each batch of _dependency manager_ lines evaluated.